File: data/src/main/java/com/popularmovies/vpaliy/data/utils/SchedulerProvider.java
package com.popularmovies.vpaliy.data.utils;

import android.support.annotation.NonNull;

import rx.Scheduler;
import rx.android.schedulers.AndroidSchedulers;
import rx.schedulers.Schedulers;

public class SchedulerProvider {

    @NonNull
    public Scheduler computation() {
        return Schedulers.computation();
    }

    @NonNull
    public Scheduler io() {
        return Schedulers.io();
    }

    @NonNull
    public Scheduler multi(){
        return Schedulers.newThread();
    }

    @NonNull
    public Scheduler ui() {
        return AndroidSchedulers.mainThread();
    }
}

File: domain/src/main/java/com/popularmovies/vpaliy/domain/executor/SchedulerProvider.kt
package com.popularmovies.vpaliy.domain.executor

import io.reactivex.Scheduler
import io.reactivex.android.schedulers.AndroidSchedulers
import io.reactivex.schedulers.Schedulers
import javax.inject.Inject

class SchedulerProvider @Inject constructor(): BaseScheduler {
  override fun io():Scheduler= Schedulers.io()
  override fun ui(): Scheduler=AndroidSchedulers.mainThread()
}